Require name for workflows on save, set default to Unnamed Workflow

ahmedhamidawan opened the pull request:
Also, use `Toast` instead of modal for empty name, as well as `Toast.success` on newly created workflow
Fixes #16998 
Fixes #16997 

| Before |
| ------ |
| <video src="https://github.com/galaxyproject/galaxy/assets/78516064/ebea7385-cda0-4bed-b4a8-e723e10d8476"> |

| After |
| ------ |
| <video src="https://github.com/galaxyproject/galaxy/assets/78516064/d4068e2f-ca0d-4bbd-b33c-144961caaf4b"> |

~~_To Do:_
_- Also require name for Workflow `update` method (`PUT /api/workflows/{id}`) like we do for the `create` method (`PUT /api/workflow/create`) already._~~ Done ✅ 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
client: require name for workflows on save, set default Unnamed

Also, use `Toast` instead of modal for empty name, as well as
`Toast.success` on newly created workflow

mvdbeek commented on lib/galaxy/webapps/galaxy/api/workflows.py:
```diff
@@ -472,6 +472,8 @@ def update(self, trans: GalaxyWebTransaction, id, payload, **kwds):
             new_workflow_name = workflow_dict.get("name")
             old_workflow = stored_workflow.latest_workflow
             name_updated = new_workflow_name and new_workflow_name != stored_workflow.name
+            if not name_updated:
```
That doesn't look right, you should be able to update e.g. only the hidden status. Move that check to line 479 and also check in https://github.com/mvdbeek/galaxy/blob/c6435f7ea2b5773fcb6fc3be6bef37b5ea4ce2ad/lib/galaxy/managers/workflows.py#L692 ?

ahmedhamidawan replied:
Ah my bad, yes this is in the wrong place entirely, moving it to the appropriate place(s)

ahmedhamidawan committed:
backend: throw error if name is empty for `PUT /api/workflows/{id}`

This is handled at the frontend in the editor to prevent this,
but still add additional checks in backend

ahmedhamidawan committed:
add api test for update workflow with empty name

martenson reviewed (approved):
Good improvement. Thanks @ahmedhamidawan !
